ntoskrnl/ex/locale.c

```diff
@@ -114,6 +116,12 @@ ExpSetCurrentUserUILanguage(IN PCWSTR MuiName,
     NTSTATUS Status;
     PAGED_CODE();
 
+    /* Check that the passed language ID is not bogus */
+    if (LanguageId & BOGUS_ID)
```

Comment:
So an ID of `0x1000` would be considered bogus as well? (your tests above don't check for that; you just tested with values that are in a sense negative.)

Reply:
It would pass since it is not within the bounds of a bogus ID but it will eventually fail later on because an ID of `0x1000` does not correspond to any defined existing locale. In this case the caller who gives in such an ID does not know what is he looking for.